atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java

```diff
@@ -1805,13 +1805,13 @@ public void close() {
      * @throws AtlasDbDependencyException if fewer than a quorum of Cassandra nodes are reachable.
      */
     @Override
-    public void addGarbageCollectionSentinelValues(TableReference tableRef, Iterable<Cell> cells) {
+    public void addGarbageCollectionSentinelValues(TableReference tableRef, Collection<Cell> cells) {
         try {
             final Value value = Value.create(PtBytes.EMPTY_BYTE_ARRAY, Value.INVALID_VALUE_TIMESTAMP);
             cellValuePutter.putWithOverriddenTimestamps(
                     "addGarbageCollectionSentinelValues",
                     tableRef,
-                    Iterables.transform(cells, cell -> Maps.immutableEntry(cell, value)));
+                    cells.stream().map(cell -> Maps.immutableEntry(cell, value)).collect(Collectors.toList()));
```

Comment:
Do we have allocation & performance test results comparing before & after? I'm a little concerned about materializing the intermediate collection in places like this and we may need to use Guava lazy transforms instead:

```suggestion
                    Collections2.transform(cells, cell -> Maps.immutableEntry(cell, value)));
```